### DISCUSSION TITLE
Latest Manjaro python upgrade broke autotiling, flashfocus, nwg-wrapper?

### DISCUSSION THREAD

**Discussion by heapifyman** [Manjaro-Sway/manjaro-sway#525] [Q&A] [answered]
[Latest Manjaro upgrade](https://forum.manjaro.org/t/stable-update-2023-06-04-kernels-gnome-44-1-plasma-5-27-5-python-3-11-toolchain-firefox/141610) seems to have switched to python 3.11 as default and that seems to have broken autotiling, flashfocus, nwg-wrapper - see error message for autotiling below.

Probably similar problem as in https://github.com/manjaro-sway/manjaro-sway/issues/216#issuecomment-1007738101 and https://github.com/manjaro-sway/manjaro-sway/issues/213, and I will have to wait for updated versions of those packages?

Or is there something I can do?

Re-installing or following the [instructions in the announcement](https://forum.manjaro.org/t/stable-update-2023-06-04-kernels-gnome-44-1-plasma-5-27-5-python-3-11-toolchain-firefox/141610#additional-info-1) did not help.

```
Traceback (most recent call last):
  File "/usr/lib/python3.11/importlib/metadata/__init__.py", line 563, in from_name
    return next(cls.discover(name=name))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/autotiling", line 33, in <module>
    sys.exit(load_entry_point('autotiling==1.8', 'console_scripts', 'autotiling')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/bin/autotiling", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
                       ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/importlib/metadata/__init__.py", line 981, in distribution
    return Distribution.from_name(distribution_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/importlib/metadata/__init__.py", line 565, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for autotiling
```

**Comment by boredland** [comment]
Hi! That is a sad, but regular issue with all python packages in the ArchLinux world.

For autotiling I decided to switch to autotiling-rs to prevent that happening in the future.

**Reply by heapifyman** [reply]
If I understand correctly, last time it was sufficient to provide a new "version" of the affected packages by increasing `pkgrel` and rebuilding? Would you do that? Is there anything I could help with? Or is the autotiling package now "deprecated" and won't be updated anymore?

Regarding autotiling-rs, it has a slightly annoying bug - see #522.  That's why I switched back to autotiling. Apologies if that is not the right place. Should I report that in the autotiling-rs repo instead?

**Reply by boredland** [reply]
Will try to rebuild all of them. I tried a few days ago with lots of errors.

- [ ] autotiling
- [ ] nwg-wrapper
- [ ] flashfocus

**Reply by boredland** [reply]
Triggered rebuilds for all of them. Can you try and reinstall? If it works I'll probably bump the pkgrel anyway.

**Reply by heapifyman** [reply]
autotiling and nwg-wrapper seem to be working again, thanks.

flashfocus still reports an error:

```
Traceback (most recent call last):
  File "/usr/bin/flashfocus", line 5, in <module>
    from flashfocus.cli import cli
ModuleNotFoundError: No module named 'flashfocus'
```

It still seems to install to `/usr/lib/python3.10`:

```
l /usr/lib/python3.10/site-packages                                                                                                           
...
drwxr-xr-x 1 root root 378 Jun  7 00:14 flashfocus
drwxr-xr-x 1 root root 110 Jun  7 00:14 flashfocus-2.3.1.dist-info
```

Maybe I don't have the latest version yet?

```
pacman -Qi flashfocus 
...
Build Date      : Di 16 Mai 2023 09:19:53 CEST
Install Date    : Mi 07 Jun 2023 00:14:49 CEST
...
```

**Reply by boredland** [reply, marked as answer]
bumped flashfocus, pls give the new version a try! I will now bump pkgrels for autotiling and nwg-wrapper to ensure this lands on all machines.

**Reply by heapifyman** [reply]
New version flashfocus `2.3.1-2` is working fine. Thanks 🙏🏽
